docs/DeveloperGuide.md

```diff
@@ -5,9 +5,262 @@
+
+The `DeleteCommand` class implements the `Command` interface and is responsible for deleting existing items in the
+inventory. A DeleteCommand instance is created when calling the `parseDeleteCommand` method called in Parser class.
+This method parses the input and ensures that the command parameter (item name) exists in the inventory. The `execute()`
+method in the class will call the `delete` method from `Inventory` class to remove the item. It will then execute
+the `saveData` method from `FileManager` class to save changes to the inventory.
+
+#### Dependencies
+- `Inventory`: Checking and deleting item from inventory
+- `Ui`: To notify the user about the successful execution of `DeleteCommand`
+
+The following sequence diagram shows the execution of a DeleteCommand<br>
+![DeleteCommandSequence](uml-diagrams/DeleteCommandSequence.png)
+
+1. The `SuperTracker` class calls the `execute` method of `DeleteCommand`
+2. The `contains` method of `Inventory` to check if the item exists in the inventory
+3. If item exists, the `delete` method of `Inventory` is called to remove the item from inventory
+4. Subsequently, the `deleteCommandSuccess` method of `Ui` is called to print the delete message
+5. The `saveData` method of `FileManager` is called to save the change to the text file containing inventory information
+
+### Find Command
+The following is a class diagram of the FindCommand and its relevant dependencies<br>
+![FindCommandClass](uml-diagrams/FindCommandClass.png)
+
+The `FindCommand` class implements the `Command` interface and is responsible for searching for items in the inventory by name.
+A FindCommand instance is created by the `parseFindCommand` method called by `Parser`, which ensures that the provided parameter (name) is valid.
+
+#### Dependencies
+- `Inventory`: For getting the list of items in the inventory
+- `Ui`: To notify the user about the successful execution of `FindCommand`
+
+The following sequence diagram shows the execution of a FindCommand<br>  
+![FindCommandSequence](uml-diagrams/FindCommandSequence.png)
+
+1. The `SuperTracker` class calls the `execute` method of `FindCommand`
+2. A boolean variable `isFound` is assigned a false value 
+3. The `getItems` method of the `Inventory` class is called to get the list of items in the inventory
+4. The `FindCommand` class will loop through each item of the list of items
+5. On every iteration, the item will be checked if it contains the word that is to be found
+6. If the item contains the word, the `foundItem` method of the Ui class is called and the `isFound` variable is assigned a true value
+7. After the loop ends, if the `isFound` variable is still false, the `noItemFound` method of the Ui class is called to notify that no item has been found containing the word
+
+### List Command
+The following is a class diagram of the FindCommand and its relevant dependencies<br>
+![ListCommandClass](uml-diagrams/ListCommandClass.png)
```

Comment:
Class diagram is too complicated. I think it can be broken down into smaller parts in other diagrams and displayed when those parts are being explained.